CRM/Utils/Check/Component/DedupeRules.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+class CRM_Utils_Check_Component_DedupeRules extends CRM_Utils_Check_Component {
+
+  /**
+   * Get dedupe rules, grouped by contact type, for a specific usage
+   *
+   * @param string $used (Supervised, Unsupervised, General)
+   * @return string[]
+   */
+  private static function getContactTypesForRule($used) {
+    $dedupeRules = \Civi\Api4\DedupeRuleGroup::get()
+      ->addSelect('contact_type')
+      ->setGroupBy([
+        'contact_type',
+      ])
+      ->addWhere('used', '=', $used)
+      ->execute();
+
+    $types = [];
+    foreach ($dedupeRules as $rule) {
+      if (!in_array($rule, $types)) {
+        $types[] = $rule['contact_type'];
+      }
+    }
+    return $types;
+  }
+
+  /**
+   * Returns an array of missing expected contact types
+   *
+   * @param string[] $rules
+   * @return array
+   */
+  private static function getMissingRules($rules) {
+    $types = ['Organization', 'Household', 'Individual'];
```

Comment:
```suggestion
    $types = CRM_Contact_BAO_ContactType::basicTypes();
```
Because not all contact types are enabled on every install.

Reply:
Using the `basicTypes` method seems much better. I didn't know that function existed.

However, I just want to note that inactive contact types are still shown on the "Find and Merge Duplicate Contacts" page, and I don't believe disabling a contact type stops CiviCRM from using it in 100% of cases. So I half wonder if its safest to ask users to ensure dedupe rules are correctly configured, even for the contact types they're not using? Bit in two minds on this, wanted to throw it out there for discussion.

Reply:
> I don't believe disabling a contact type stops CiviCRM from using it in 100% of cases.

Right, because not everyone knows that function exists :). There are still some hardcoded `['Organization', 'Household', 'Individual']` arrays in the codebase which need replacing.

The solution IMO is to keep moving forward in that direction (using the function, not hardcoded lists) and follow-up this PR with a fix to the "Find and Merge Duplicate Contacts" screen to also respect enabled contact types.